server/apps/immich/src/config/asset-upload.config.ts

```diff
@@ -42,6 +43,12 @@ function destination(req: Request, file: Express.Multer.File, cb: any) {
     return cb(new UnauthorizedException());
   }
 
+  const user = req.user as AuthUserDto;
+
+  if (user.isPublicUser && !user.isAllowUpload) {
+    return cb(new UnauthorizedException());
+  }
```

Comment:
Nice! Although, I think should technically be forbidden, since the use _is_ authenticated, they're just not allowed to do this. So 401 = no/invalid authentication, 403 = has auth, which is valid, but doesn't grant permission for the operation.